compiler/src/dotty/tools/dotc/consumetasty/ConsumeTasty.scala

```diff
@@ -18,23 +23,33 @@ object ConsumeTasty {
         new TastyFromClass(tastyConsumer)
     }
 
-    val currentClasspath = getCurrentClasspath(getClass.getClassLoader)
-    import java.io.File.{ pathSeparator => sep }
+    val currentClasspath = classpathFromClassloader(getClass.getClassLoader)
     val args = "-from-tasty" :: "-Yretain-trees" :: "-classpath" :: s"$classpath$sep$currentClasspath" :: classes
     (new Consume).process(args.toArray)
   }
 
-  private def getCurrentClasspath(cl: ClassLoader): String = {
-    val classpath0 = System.getProperty("java.class.path")
-    cl match {
-      case cl: URLClassLoader =>
-        // Loads the classes loaded by this class loader
-        // When executing `run` or `test` in sbt the classpath is not in the property java.class.path
-        import java.nio.file.Paths
-        val newClasspath = cl.getURLs.map(url => Paths.get(url.toURI).toString)
-        newClasspath.mkString("", java.io.File.pathSeparator, if (classpath0 == "") "" else java.io.File.pathSeparator + classpath0)
-      case _ => classpath0
-    }
+  /** Attempt to recreate a classpath from a classloader.
+   *
+   *  BEWARE: with exotic enough classloaders, this may not work at all or do
+   *  the wrong thing.
+   */
+   private def classpathFromClassloader(cl: ClassLoader): String = {
+    @tailrec
+    def loop(cl: ClassLoader, suffixClasspath: String): String =
+      cl match {
+        case cl: URLClassLoader =>
+          val updatedClasspath = cl.getURLs
+            .map(url => Paths.get(url.toURI).toAbsolutePath.toString)
+            .mkString(
+              "",
+              File.pathSeparator,
+              if (suffixClasspath.isEmpty) "" else File.pathSeparator + suffixClasspath
```

Comment:
The current version in #7953 has been updated. You should take the current version.

Reply:
Hi, Nicolas.  I'm not sure what you mean.  I looked at #7953, which is now merged into master.  ConsumeTasty.scala in master still looks like the old way of getting the class path, not @smarter's new way.  What change is requested?  Thx.

Reply:
I meant, to copy the code in https://github.com/lampepfl/dotty/pull/7953/files#diff-0cf1d278cf2ee8dca0cb2b7bcb62348aR75

Reply:
Should we take this as a sign to refactor into one method?

Reply:
Yes, but I would leave it for a future PR. I have a bigger refactoring coming that will address that.

Reply:
I opened #8020 for this refactoring.